packages/commerce-sdk-react/src/hooks/useTrustedAgent.ts

```
@@ -0,0 +1,178 @@
+                popup?.focus()
+            }
+        }
+
+        checkPopupState()
+        intervalId = setInterval(checkPopupState, 1000)
+    })
+}
+
+/**
+ * A hook to return trusted agent state.
+ *
+ * @group Helpers
+ * @category Shopper Authentication
+ *
+ */
+const useTrustedAgent = (): UseTrustedAgent => {
+    const auth = useAuthContext()
+    const [isAgent, setIsAgent] = useState(false)
+    const [agentId, setAgentId] = useState('')
+    const [loginId, setLoginId] = useState('')
+
+    const authorizeTrustedAgent = useMutation(auth.authorizeTrustedAgent.bind(auth))
+    const loginTrustedAgent = useMutation(auth.loginTrustedAgent.bind(auth))
+    const logoutTrustedAgent = useMutation(auth.logout.bind(auth))
+
+    const login = useCallback(
+        async (loginId?: string, usid?: string, refresh = false): Promise<TokenResponse> => {
+            if (!onClient()) {
+                throw new Error(
+                    'Something went wrong, this client side method is invoked on the server.'
+                )
+            }
+
+            const {url, codeVerifier} = await authorizeTrustedAgent.mutateAsync({loginId})
+            const {code, state} = await createTrustedAgentPopup(url, refresh)
+            return await loginTrustedAgent.mutateAsync({
+                loginId,
+                code,
+                codeVerifier,
+                state,
+                usid
+            })
+        },
+        [auth]
+    )
+
+    const logout = useCallback(async () => {
+        return await logoutTrustedAgent.mutateAsync()
+    }, [auth])
+
+    useEffect(() => {
+        auth.registerTrustedAgentRefreshHandler(login)
+    }, [auth])
+
+    useEffect(() => {
+        try {
+            const {isAgent, agentId, loginId} = auth.parseSlasJWT(auth.get('access_token'))
+            setIsAgent(isAgent)
+            setAgentId(agentId || '')
+            setLoginId(loginId)
+        } catch (e) {
+            /* here to catch invalid jwt errors */
```

seems like error is swallowed here, how do we handle errors here?